src/strings.rs

```diff
@@ -17,11 +17,49 @@ declare_lint! {
     "Warn on `x = x + ..` where x is a `String`"
 }
 
-#[derive(Copy,Clone)]
+declare_lint! {
+	pub STRING_ADD,
+	Allow,
+	"Warn on `x + ..` where x is a `String`"
+}
+
+#[derive(Copy, Clone)]
 pub struct StringAdd;
 
 impl LintPass for StringAdd {
     fn get_lints(&self) -> LintArray {
+        lint_array!(STRING_ADD)
+    }
+
+    fn check_expr(&mut self, cx: &Context, e: &Expr) {
```

Comment:
Yeah, the disabling is done in individual lints, not passes. The passes are unconditionally all run simultaneously (It's a single AST visit)